tests/interfaces/default_qubit_2_integration/test_jax_default_qubit_2.py

```diff
@@ -521,7 +588,8 @@ def cost(x, y):
             ops = [qml.RX(x, wires=0), qml.RY(y, 1), qml.CNOT((0, 1))]
             m = [qml.expval(qml.PauliZ(0)), qml.probs(wires=1)]
             tape = qml.tape.QuantumScript(ops, m, shots=shots)
-            return jnp.hstack(execute([tape], device, **execute_kwargs)[0])
+            res = qml.execute([tape], device, **execute_kwargs)[0]
+            return jnp.hstack(jax.tree_util.tree_leaves(res))
```

Comment:
can users get something to this effect without manually calling `tree_leaves`? or rather, why is it necessary?

Reply:
With a shot vector, the unstacked `res` is:

```
((Array(0.855, dtype=float32), Array([0.8319, 0.1681], dtype=float32)),
 (Array(0.854, dtype=float32), Array([0.8472, 0.1528], dtype=float32)))
```

Because of the extra level of nesting, `hstack` can't be called on that nested data structure directly.

By adding a call to `tree_leaves`, I flatten out the nested structure of the result, and allow it to be used with `jnp.hstack`.

Honestly, I'm not even quite sure why we are using `hstack` at all.

Reply:
I was worried that we're testing something users can't really do, and I guess that isn't the case. it's more a question of would they than could they, and I suppose now we know that if they want to flatten and hstack their results, they could! that's fine with me

Reply:
And technically the code works fine without any flattening and stacking.
<img width="747" alt="Screenshot 2023-11-15 at 4 27 44 PM" src="https://github.com/PennyLaneAI/pennylane/assets/6364575/79171674-a337-4e62-a28f-1480725fdbf2">